src/libsyntax_ext/deriving/mod.rs

```diff
@@ -96,6 +96,36 @@ fn expand_derive(cx: &mut ExtCtxt,
             let mut found_partial_eq = false;
             let mut found_eq = false;
 
+            // This span is **very** sensitive and crucial to
+            // getting the stability behavior we want. What we are
+            // doing is marking `#[structural_match]` with the
```

Comment:
If I understand correctly, we could also be marking `#[derive_*]` with this span -- the comment should probably reflect that.